docs/implementation.md

```diff
@@ -216,7 +216,7 @@ GET /transactions?since=170&transaction-time-hash=00000000000000000000100158f474
       "transactionNumber": "A monotonically increasing number (need NOT be by 1) that identifies a Sidtree transaction.",
       "transactionTime": "The logical blockchain time this transaction is anchored. Used for protocol version selection.",
       "transactionTimeHash": "The hash associated with the transaction time.",
-      "anchorFileHash": "Hash of the anchor file of this transaction."
+      "anchorString": "The string written to the blockchain for this transaction."
```

Comment:
The size limitations of this string should be documented.

Reply:
32 byte utf-8 string seems correct here.

In practice this string may be a protocol specific encoding of a multihash.

32 bytes is not enough for much more than a hash or a uuid.

I worry about the security issues arising from weakening this type.

For example, a uuid might work fine to resolve the anchor file, but provides no built in integrity check.

There is nothing preventing an implementer from returning content for an anchorFile that does not match its hash, but now there will be no way of knowing that is happening from looking at the ledger.

I suggest renaming CAS as well, if content addressing is not longer a core requirement.

in element, we use the term "storage" specifically to eliminate this confusion, we support ipfs, but also centralized databases that can return anchorFiles which don't match the anchorFileHash that was written to the ledger.

For GDPR Compliance, private consortium and development purposes, I think its fine if some anchorFileHashes are not resolved over IPFS / SWARM / Public P2P Storage, but I think that content addressing is a valuable feature of the protocol, and we should try and default to integrity checked storage interfaces.

Reply:
Thanks @OR13,

Updated protocol documentation to include concrete parameter values used in the latest protocol version.

Reply:
Hi @OR13, I did not refresh to get you second comment. You are right that the length of the anchor string will be protocol version specific. @csuwildcat and I had a lengthy chat on what type of checks should be done to prevent abusers from supplying false/bogus data for the anchor string, and we arrived at the conclusion that we can't really prevent abusers from doing it, so we will let protocol version dictate what validation is done at the observing time (remember strict checks are done at resolution time). To be clear, we see CAS still being integral part of the design, meaning the implementer should definitely fail the content if the hash does not match, this change is really to address the case if in the future we write more information than just the anchor file hash to the blockchain.

@csuwildcat feel free to add your 2 cents.